drivers/adc/max9611pmb/max9611.c

```diff
@@ -0,0 +1,465 @@
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief Configures the internal MUX settings
+ * @param dev 		  - The device structure.
+ * @param mux 		  - MUX configuration
+ * @return            - 0 if successful, error otherwise.
+*******************************************************************************/
+int max9611_set_mux(struct max9611_dev *dev, enum max9611_mux_conf mux)
+{
+	switch (mux) {
+	case MAX9611_CONF_SENSE_1X:
+		dev->capture_type = MAX9611_DATA_CSA;
+		break;
+	case MAX9611_CONF_SENSE_4X:
+		dev->capture_type = MAX9611_DATA_CSA;
+		break;
+	case MAX9611_CONF_SENSE_8X:
+		dev->capture_type = MAX9611_DATA_CSA;
+		break;
+	case MAX9611_CONF_IN_COM_MODE:
+		dev->capture_type = MAX9611_DATA_RSP;
+		break;
+	case MAX9611_CONF_OUT_ADC:
+		dev->capture_type = MAX9611_DATA_OUT;
+		break;
+	case MAX9611_CONF_SET_ADC:
+		dev->capture_type = MAX9611_DATA_SET;
+		break;
+	case MAX9611_CONF_TEMP:
+		dev->capture_type = MAX9611_DATA_TMP;
+		break;
+	default:
+		dev->capture_type = MAX9611_DATA_CSA;
+		break;
+	}
+
+	return max9611_reg_update(dev, MAX9611_CTR1, mux, MAX9611_MUX_MASK);
+}
+
+/***************************************************************************//**
+ * @brief Get the internal MUX settings
+ * @param dev 		  - The device structure.
+ * @param mux 		  - Pointer to store the obtained mux config
+ * @return            - 0 if successful, error otherwise.
+*******************************************************************************/
+int max9611_get_mux(struct max9611_dev *dev, enum max9611_mux_conf *mux)
+{
+	int ret;
+	uint8_t temp;
+
+	ret = max9611_read(dev, MAX9611_CTR1, 1, &temp);
+	if (ret)
+		return ret;
+
+	*mux = temp;
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief Device shutdown or power up
+ * @param dev 		  - The device structure.
+ * @param is_shdn	  - Set to true to shutdown device, false otherwise
+ * @return            - 0 if successful, error otherwise.
+*******************************************************************************/
+int max9611_shutdown(struct max9611_dev *dev, bool is_shdn)
+{
+	if (is_shdn)
+		return max9611_reg_update(dev, MAX9611_CTR1, no_os_field_prep(MAX9611_SHDN_MASK,
+					  1), MAX9611_SHDN_MASK);
+	else
+		return max9611_reg_update(dev, MAX9611_CTR1, 0, MAX9611_SHDN_MASK);
+}
+
+/***************************************************************************//**
+ * @brief Configure LR bit operation
+ * @param dev 		  - The device structure.
+ * @param is_normal	  - Set true for normal operation. False otherwise (refer to LR bit in datasheet)
+ * @return            - 0 if successful, error otherwise.
+*******************************************************************************/
+int max9611_set_lr(struct max9611_dev *dev, bool is_normal)
+{
+	if (is_normal)
+		return max9611_reg_update(dev, MAX9611_CTR1, 0, MAX9611_LR_MASK);
+	else
```

Comment:
remove the `else`. Same for the `max9611_shutdown()`.

Reply:
Retained in latest commit. Boolean input used to dictate operation to perform instead of having two different operations depending on target configuration.

Reply:
This is what I mean:
```
if (...)
    return reg_update()
return reg_update()
```

Has the same behavior, but there is no `else`. This is a pattern we usually use.

Reply:
Implemented in latest commit.